adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -48,191 +43,138 @@ @implementation GADMAdapterUnityRewardedAd
-                                                                        delegate:self];
+    if (![UnityAds isReady:_placementID]) {
+        NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
+                                                                     GADMAdapterUnityErrorShowAdNotReady, @"Failed to show Unity Ads rewarded video.");
+        [_adEventDelegate didFailToPresentWithError:error];
+        return;
+    }
+    [_adEventDelegate willPresentFullScreenView];
+    [UnityAds show:viewController placementId:_placementID];
 }
 
 #pragma mark GADMAdapterUnityDataProvider Methods
 
 - (NSString *)getGameID {
-  return _gameID;
+    return _gameID;
 }
 
 - (NSString *)getPlacementID {
-  return _placementID;
+    return _placementID;
 }
 
 - (void)didFailToLoadWithError:(nonnull NSError *)error {
-  _adLoadCompletionHandler(nil, error);
+    _adLoadCompletionHandler(nil, error);
 }
 
 #pragma mark - Unity Delegate Methods
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(nonnull NSString *)message {
-  if (error == kUnityAdsErrorNotInitialized) {
-    if (_adLoadCompletionHandler) {
-      NSError *errorWithDescription =
-          GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-      _adLoadCompletionHandler(nil, errorWithDescription);
-      _adLoadCompletionHandler = nil;
-    }
-  } else {
     if (_adEventDelegate) {
-      NSError *errorWithDescription =
-          GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-      [_adEventDelegate didFailToPresentWithError:errorWithDescription];
+        NSError *errorWithDescription =
+        GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
+        [_adEventDelegate didFailToPresentWithError:errorWithDescription];
     }
-  }
 }
 
 - (void)unityAdsDidFinish:(nonnull NSString *)placementID
           withFinishState:(UnityAdsFinishState)state {
-  if (![placementID isEqualToString:_placementID]) {
-    return;
-  }
-
-  if (state == kUnityAdsFinishStateCompleted) {
-    [_adEventDelegate didEndVideo];
-
-    // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
-    // amount of 1. Publishers using this adapter should override the reward on the AdMob
-    // front-end.
-    GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
-                                                     rewardAmount:[NSDecimalNumber one]];
-    [_adEventDelegate didRewardUserWithReward:reward];
-  } else if (state == kUnityAdsFinishStateError) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorFinish,
-        @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
-    [_adEventDelegate didFailToPresentWithError:error];
-  }
-
-  [_adEventDelegate willDismissFullScreenView];
-  [_adEventDelegate didDismissFullScreenView];
+    if (![placementID isEqualToString:_placementID]) {
```

Comment:
Sergey has said to me that we don't need this check anymore (would apply to all below too) - @serge-14